src/CheckoutSdk/Reconciliation/ReconciliationClient.cs

```diff
@@ -7,32 +8,64 @@ public class ReconciliationClient : AbstractClient, IReconciliationClient
         private const string ReportingPath = "reporting";
         private const string PaymentsPath = "payments";
+        private const string DownloadPath = "download";
         private const string StatementsPath = "statements";
 
         public ReconciliationClient(IApiClient apiClient, CheckoutConfiguration configuration) : base(apiClient,
             configuration, SdkAuthorizationType.SecretKey)
         {
         }
 
-        public Task<ReconciliationPaymentReportResponse> QueryPaymentsReport(ReconciliationQueryPaymentsFilter filter)
+        public Task<ReconciliationPaymentReportResponse> QueryPaymentsReport(ReconciliationQueryPaymentsFilter filter,
+            CancellationToken cancellationToken = default)
         {
             CheckoutUtils.ValidateParams("filter", filter);
             return ApiClient.Query<ReconciliationPaymentReportResponse>(BuildPath(ReportingPath, PaymentsPath),
-                SdkAuthorization(), filter);
+                SdkAuthorization(), filter, cancellationToken);
         }
 
-        public Task<ReconciliationPaymentReportResponse> SinglePaymentReport(string paymentId)
+        public Task<ReconciliationPaymentReportResponse> SinglePaymentReport(string paymentId,
+            CancellationToken cancellationToken = default)
         {
             CheckoutUtils.ValidateParams("paymentId", paymentId);
             return ApiClient.Get<ReconciliationPaymentReportResponse>(BuildPath(ReportingPath, PaymentsPath, paymentId),
-                SdkAuthorization());
+                SdkAuthorization(), cancellationToken);
         }
 
-        public Task<StatementReportResponse> QueryStatementsReport(QueryFilterDateRange filter)
+        public Task<StatementReportResponse> QueryStatementsReport(QueryFilterDateRange filter,
+            CancellationToken cancellationToken = default)
         {
             CheckoutUtils.ValidateParams("filter", filter);
             return ApiClient.Query<StatementReportResponse>(BuildPath(ReportingPath, StatementsPath),
-                SdkAuthorization(), filter);
+                SdkAuthorization(), filter, cancellationToken);
+        }
+
+        public Task<string> RetrieveCsvPaymentReport(QueryFilterDateRange filter,
+            CancellationToken cancellationToken = default)
+        {
+            CheckoutUtils.ValidateParams("filter", filter);
+            return ApiClient.Query<string>(BuildPath(ReportingPath, PaymentsPath, DownloadPath), SdkAuthorization(),
+                filter,
+                cancellationToken);
+        }
+
+        public Task<string> RetrieveCsvSingleStatementReport(string statementId,
+            CancellationToken cancellationToken = default)
+        {
+            CheckoutUtils.ValidateParams("statementId", statementId);
+            return ApiClient.Query<string>(
+                BuildPath(ReportingPath, StatementsPath, statementId, PaymentsPath, DownloadPath), SdkAuthorization(),
+                new QueryFilterDateRange(),
```

Comment:
If we are creating a `new QueryFilterDateRange()` isn't more appropriate to use `get` operation only?

Reply:
Good question. This was more a workaround to avoid the validation on `Serialize()` line 24. Will try a different approach adding a null check on `ApiClient` if this reference is null.